pandas/_testing/__init__.py

```diff
@@ -285,11 +286,17 @@ def box_expected(expected, box_cls, transpose: bool = True):
         else:
             expected = pd.array(expected, copy=False)
     elif box_cls is Index:
-        expected = Index(expected)
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", "Dtype inference", category=FutureWarning)
```

Comment:
these warnings seem like they might add up in test runtime.  might be more performant to pass dtype=expected.dtype to the constructors?  (i guess that is making a not-enforced assumption about what expected is?)

Reply:
we sometimes rely on the inference behaviour, that's why I kept this as is for now, I don't want to screw up testing before we enforce the deprecation